vpr/src/route/rr_graph_storage.cpp

```diff
@@ -709,3 +709,68 @@ t_rr_graph_view t_rr_graph_storage::view() const {
         vtr::make_const_array_view_id(edge_dest_node_),
         vtr::make_const_array_view_id(edge_switch_));
 }
+
+void t_rr_graph_storage::reorder(const vtr::vector<RRNodeId, RRNodeId>& order,
+                                 const vtr::vector<RRNodeId, RRNodeId>& inverse_order) {
+    {
+        auto old_node_storage = node_storage_;
```

Comment:
Do you need to have 2x copies of the data for the re-order?

Reply:
Yes. Note that `node_storage_` is written in a different pattern than `old_node_storage` is read. I believe I made this error initially before realizing I needed two copies. The anonymous block scope is used to allow the copy to be discarded as soon as possible.

Reply:
Let me rephrase.  Could you write a version of this loop that doesn't required 2x data?  And instead only needs a temporary?

I agree that the obvious reordering algorithm needs 2x memory.  Could you instead write something that consumes more runtime, but doesn't have the 2x memory hit?  As a concrete example of a solution, a non-stable sorting of the `zip(order, node_storage),sort(key=lambda x: x[0])` sorting on the order would achieve this.  There are other options here.

Reply:
It might be possible to trade memory usage for runtime, but I don't think it's worthwhile.

Reply:
We have rr-graphs that consume 22-44 GBs, and doubling that overhead doesn't seem worth avoiding?

Reply:
This is an optional feature that can consume more memory, but is not enabled by default.

At ~44GB and a `t_rr_node_data` size of 16 bytes, that means ~2.75 billion nodes, so ~2^(31.3). So, it might take 31x more time to compute. But thats ignoring the cache. Sorting will likely be less cache-friendly.

I have not analyzed this trade-off, because that would take significant time to perform, requiring me to write and test the other way of shuffling nodes, which will likely not pan out. I choose not to do this, unless justified.

Nonetheless, my quick calculations show that this would probably not be a good tradeoff.

Also note that effort was taken to copy things in stages, so the memory requirement is less than 2x the entire graph.

Reply:
I've provided data below.